### PR TITLE
Update maintainers guide

### PR DESCRIPTION
**Pull request by communikit**
- Removed TOC
- Minor edits

See http://b/181807514

### CODE REVIEW

File: maintaining/maintainers-guide.md [hunk 1]
```diff
@@ -31,7 +23,7 @@ manage aspects of the open source project. These are:
 * **Release Process**: Manage Bazel's release process.
 * **Green Team**: Grow a healthy ecosystem of rules and tools.
 * **Developer Experience Gardeners**: Encourage external contributions, review issues
-  and pull requests, make our development workflow more open, and shift the team 
+  and pull requests, make our development workflow more open, and shift the team
```
Comment (aiuto) [inline maintaining/maintainers-guide.md:26]: The "and shift the team towards github-first development" 
should be dropped. That is not an objective of that group.
It is more of an overall aspirational goal for the project.

**Review by aiuto — APPROVED** [review]

File: maintaining/maintainers-guide.md (hunk 1 above)
Comment (communikit) [reply maintaining/maintainers-guide.md]: Took the wording out of the paragraph.

**Review by aiuto — APPROVED** [review]

File: maintaining/maintainers-guide.md [hunk 2]
```diff
@@ -31,8 +23,7 @@ manage aspects of the open source project. These are:
 * **Release Process**: Manage Bazel's release process.
 * **Green Team**: Grow a healthy ecosystem of rules and tools.
 * **Developer Experience Gardeners**: Encourage external contributions, review issues
-  and pull requests, make our development workflow more open, and shift the team 
-  towards GitHub-first development.
+  and pull requests, and make our development workflow more open.
 
 ## Releases
```
Comment (floriographygoth) [inline maintaining/maintainers-guide.md:28]: Should there be parallel sections for the different maintainer teams? It looks like Releases and CI are the first two and everything else is part of the Gardeners.

**Review by floriographygoth — APPROVED** [review]
We can follow up with another PR on editorial changes after a conversation, but these changes LGTM.

**Pull request merged**